### PR TITLE
add Polish translations

### PR DESCRIPTION
**Pull request by jrushlow**
mirrors https://github.com/SymfonyCasts/reset-password-bundle/pull/149

Context:

`This link will expire in 1 year` which in Polish is `Link wygaśnie w ciągu 1 roku`
`This link will expire in 3 years` which in Polish is `Link wygaśnie w ciągu 3 lat`
`This link will expire in 5 years` which in Polish is `Link wygaśnie w ciągu 5 lat`
`This link will expire in 6 years` which in Polish is `Link wygaśnie w ciągu 6 lat`
`This link will expire in 17 years` which in Polish is `Link wygaśnie w ciągu 17 lat`

the same context applies for `month|months`, `day|days`, `hour|hours`, `minute|minutes`

### CODE REVIEW

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (mwadon) [inline src/Resources/translations/VerifyEmailBundle.pl.xlf:7]: I'm not sure is it correct translation. Can you write some sentence with this translation to understand your intention?

**Review by mwadon — CHANGES_REQUESTED** [review]

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (jrushlow) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf:7]: `This link will expire in 1 year` || `This link will expire in 3 years`

I'm not fluent in Polish, so please feel free to correct this if it's wrong. Thanks @mwadon

Comment (weaverryan) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf:7]: Yes! In English, it would look like this:

> This link will expire in {{ resetToken.expirationMessageKey|trans(resetToken.expirationMessageData, 'ResetPasswordBundle') }}.

Where the `{{ }}` part will be the translation - for example "1 year" or "3 years"` in English.

Thanks!

**Comment by jrushlow** [comment]
@marmichalski any chance you could verify these translations if you have some free time? Thanks in advance!

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (mwadon) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf:7]: It's more complicated in Polish:
```
1 year - 1 rok
2,3,4 years - 2,3,4 lata
5,6,7... years - 5,6,7... lat
```

Comment (piotrdziubczynski) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf:7]: It's not as simple in Polish as in English, but... I'm working on your above example.
1 rok
2 lata, 3 lata, 4 lata
5 lat, 6 lat and so on...

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (mwadon) [inline src/Resources/translations/VerifyEmailBundle.pl.xlf:11]: The same here:
```
1 month - 1 miesiąc
2,3,4 months - 2 miesiące
5,6,7... months - 5,6,7... miesięcy
```

**Review by mwadon — CHANGES_REQUESTED** [review]

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (piotrdziubczynski) [inline src/Resources/translations/VerifyEmailBundle.pl.xlf:11]: 1 miesiąc
2 miesiące, 3 miesiące, 4 miesiące
5 miesięcy, 6 miesięcy and so on...

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (piotrdziubczynski) [inline src/Resources/translations/VerifyEmailBundle.pl.xlf:15]: 1 dzień
2 dni, 3 dni and so on...

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (piotrdziubczynski) [inline src/Resources/translations/VerifyEmailBundle.pl.xlf:19]: 1 godzinę
2 godziny, 3 godziny, 4 godziny
5 godzin, 6 godzin and so on...

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (piotrdziubczynski) [inline src/Resources/translations/VerifyEmailBundle.pl.xlf:23]: 1 minutę
2 minuty, 3 minuty, 4 minuty
5 minut, 6 minut and so on...

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (stof) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf:7]: But that's supported in Symfony already. The `pl` locale has more than 2 variants. So this should look like that: `%count% roku|%count% lata|%count% lat`

**Comment by stof** [comment]
@weaverryan any reason to use the old Symfony-specific format in the bundle rather than the ICU format ? The bundle requires  Symfony 4.4+, so it has access to the ICU format

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (stof) [inline src/Resources/translations/VerifyEmailBundle.pl.xlf:15]: you still have only 2 variants here, which is broken if Symfony decides it needs the third one

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (stof) [inline src/Resources/translations/VerifyEmailBundle.pl.xlf:23]: this one is also missing the third variant.

**Review by stof — CHANGES_REQUESTED** [review]

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (jrushlow) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf]: fixed

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (thomas2411) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf]: In this particular context as @jrushlow wrote we have:

`This link will expire in 1 year` which in Polish is `Link wygaśnie w ciągu 1 roku`
`This link will expire in 3 years` which in Polish is `Link wygaśnie w ciągu 3 lat`

As @mwadon and @piotrdziubczynski indeed in Polish it is more difficult because of conjugation but in this particular context we need only two forms as it is in genitive.

`This link will expire in 1 year` which in Polish is `Link wygaśnie w ciągu 1 roku`
`This link will expire in 3 years` which in Polish is `Link wygaśnie w ciągu 3 lat`
`This link will expire in 5 years` which in Polish is `Link wygaśnie w ciągu 5 lat`
`This link will expire in 6 years` which in Polish is `Link wygaśnie w ciągu 6 lat`
`This link will expire in 17 years` which in Polish is `Link wygaśnie w ciągu 17 lat`

So the translation `%count% roku|%count% lat` is correct in this context.

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (thomas2411) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf]: As wrote here https://github.com/SymfonyCasts/verify-email-bundle/pull/56#discussion_r577870396 only two forms are required. But if you need 3 forms to work properly, then the second and third would be the same in this context.

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (jrushlow) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf]: Thank you all for helping us out with this. I do however need to get a few more people to validate what the correct translations are in this context this before we can merge. Please don't take that the wrong way - from a maintainers point of view who does not understand the Polish dialect as I do in English, I can't say "yup this way is correct" or "nope that way is correct".. 

If anyone knows anyone who can review, please have them take a look - it is greatly appreciated. Thank you all for your time!

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (jrushlow) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf]: thanks guys!

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (thomas2411) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf]: No worries @jrushlow, I wouldn't think that way. I know that this is how it works. I would do the same. Thank you for your time developing all of this and maintaining it.

Comment (piotrdziubczynski) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf]: @thomas2411

It really depends on sentence translation. My translation (and not only my) was: `Ten link wygaśnie za ...`, so `1 rok` is correct. In your translation (`Link wygaśnie w ciągu ...`) `1 roku` is also correct. So have to take just one convention.

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (piotrdziubczynski) [inline src/Resources/translations/VerifyEmailBundle.pl.xlf:12]: Double pipe.

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (piotrdziubczynski) [inline src/Resources/translations/VerifyEmailBundle.pl.xlf:11]: Double pipe.

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (thomas2411) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf]: Indeed, that is a bummer. OK, so probably this would be a better option to have 3 different forms as it is more popular. But should be the whole sentence put in translations too so that there are no problems like this in the future? I would be for this option. Put the whole sentence int the translation to get proper context or do not put it at all and leave it up to the developer how to proceed. So just leave `getExpirationMessageData` method and skip `getExpirationMessageKey`.

Comment (mi6th) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf]: > @thomas2411
> 
> It really depends on sentence translation. My translation (and not only my) was: `Ten link wygaśnie za ...`, so `1 rok` is correct. In your translation (`Link wygaśnie w ciągu ...`) `1 roku` is also correct. So have to take just one convention.

To me

1. *wygaśnie za* is more *exact*: *wygaśnie w ciągu* sounds like a process of expiration (*expires during a year…*).
2. With the *wygaśnie za* phrase the grammar gender of the following time term is easier to express/maintain in Polish because of the accusative form that coincides with the nominative one: "rok", "miesiąc", "tydzień", "dzień" and so on have no suffixes (*za* rok => *w ciągu* rok**u**) nor fancy declesions (*za* tydzień => *w ciągu* ty**godnia**, *za* miesiąc => *w ciągu* miesiąc**a**); but: godzina [hour]: *za* godzin**ę** > *w ciągu* godzin**y**.

Comment (thomas2411) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf]: I agree, but as wrote above I recommend to use the full sentence in the translation or do not translate it at all in the bundle and leave the whole to the developer. Both solutions would be good so that we do not have to spend time on discussions like these :)

**Comment by Inscure** [comment]
It would be nice to be able to use this code already :) When are you planning to merge?

**Comment by weaverryan** [comment]
> @weaverryan any reason to use the old Symfony-specific format in the bundle rather than the ICU format ? The bundle requires Symfony 4.4+, so it has access to the ICU format

@stof - probably not :). Though I personally understand the new format less. We were kind of following the standard in Symfony itself.

@jrushlow it looks like this has a review on it - so we could use that to push things forward.

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (jrushlow) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf]: done

Comment (jrushlow) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf]: done

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (jrushlow) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf]: done

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (jrushlow) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf]: done

File: src/Resources/translations/VerifyEmailBundle.pl.xlf
Comment (jrushlow) [reply src/Resources/translations/VerifyEmailBundle.pl.xlf]: thanks! fixed

**Pull request merged**
